File: SpringBootCustomJackson/src/main/java/com/example/component/EncodedJsonComponent.java
package com.example.component;

import com.example.domain.Model;
import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.core.ObjectCodec;
import com.fasterxml.jackson.databind.*;
import org.springframework.boot.jackson.JsonComponent;
import org.springframework.util.Base64Utils;

import java.io.IOException;
import java.util.Iterator;

@JsonComponent
public class EncodedJsonComponent {

	public static class DecyptDataDeserializer extends JsonDeserializer<Model> {

		@Override
		public Class<?> handledType() {
			return Model.class;
		}

		@Override
		public Model deserialize(JsonParser p, DeserializationContext ctxt)
				throws IOException {
			ObjectCodec codec = p.getCodec();
			JsonNode node = codec.readTree(p);

			Iterator<String> it = node.fieldNames();
			while (it.hasNext()) {
				String field = it.next();
				System.out.println(field + ":" + node.get(field));
			}

			String name = new String(Base64Utils.decodeFromString(node.get("name").asText()));
			int type = node.get("type").asInt();
			System.out.println("---------------------------------------------------");
			return new Model(name, type);
		}
	}

	public static class EncyptDataSerializer extends JsonSerializer<Model> {

		@Override
		public Class<Model> handledType() {
			return Model.class;
		}

		@Override
		public void serialize(Model value, JsonGenerator json,
		                      SerializerProvider provider) throws IOException {
			json.writeStartObject();
			json.writeFieldName("name");
			json.writeString(Base64Utils.encodeToString(value.getName().getBytes()));
			json.writeFieldName("type");
			json.writeNumber(value.getType());
			json.writeEndObject();
		}

	}
}

File: SpringBootCustomJackson/src/main/kotlin/com/example/kotlin/component/EncodedJsonComponent.kt
package com.example.kotlin.component

import com.example.kotlin.domain.Model
import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.*
import org.springframework.boot.jackson.JsonComponent
import java.io.IOException
import java.util.Base64

@JsonComponent
class EncodedJsonComponent {

    class DecyptDataDeserializer : JsonDeserializer<Model>() {

        override fun handledType(): Class<*> {
            return Model::class.java
        }

        @Throws(IOException::class)
        override fun deserialize(p: JsonParser, ctxt: DeserializationContext): Model {
            val codec = p.codec
            val node = codec.readTree<JsonNode>(p)

            val it = node.fieldNames()
            while (it.hasNext()) {
                val field = it.next()
                println(field + ":" + node.get(field))
            }

            val name = String(Base64.getDecoder().decode(node.get("name").asText()))
            val type = node.get("type").asInt()
            println("---------------------------------------------------")
            return Model(name, type)
        }
    }

    class EncyptDataSerializer : JsonSerializer<Model>() {

        override fun handledType(): Class<Model> {
            return Model::class.java
        }

        @Throws(IOException::class)
        override fun serialize(value: Model, json: JsonGenerator,
                               provider: SerializerProvider) {
            json.writeStartObject()
            json.writeFieldName("name")
            json.writeString(Base64.getEncoder().encodeToString(value.name.toByteArray()))
            json.writeFieldName("type")
            json.writeNumber(value.type)
            json.writeEndObject()
        }

    }
}
